qa/tasks/cephfs/kernel_mount.py

```diff
@@ -77,28 +80,45 @@ def _run_mount_cmd(self, mntopts, check_status):
                         mountcmd_stderr.getvalue())
         log.info('mount command passed')
 
+    def _make_mount_cmd_old_or_new_style(self, style):
+        optd = {}
+        mnt_stx = ''
+        if style == 'old':
+            mnt_stx = f':{self.cephfs_mntpt}'
+            if self.client_id:
+                optd['name'] = self.client_id
+            if self.cephfs_name:
+                optd['mds_namespace'] = self.cephfs_name
+        elif style == 'new':
+            mnt_stx = f'{self.client_id}@{self.cephfs_name}={self.cephfs_mntpt}'
+        else:
+            assert 0, 'invalid syntax style!'
+        return (mnt_stx, optd)
+
     def _get_mount_cmd(self, mntopts):
         opts = 'norequire_active_mds'
-        if self.client_id:
-            opts += ',name=' + self.client_id
         if self.client_keyring_path and self.client_id:
             opts += ',secret=' + self.get_key_from_keyfile()
         if self.config_path:
             opts += ',conf=' + self.config_path
-        if self.cephfs_name:
-            opts += ",mds_namespace=" + self.cephfs_name
         if self.rbytes:
             opts += ",rbytes"
         else:
             opts += ",norbytes"
-        if mntopts:
-            opts += ',' + ','.join(mntopts)
 
         mount_cmd = ['sudo'] + self._nsenter_args
-        mount_dev = ':' + self.cephfs_mntpt
-        mount_cmd += self._mount_bin + [mount_dev, self.hostfs_mntpt, '-v',
-                                        '-o', opts]
-
+        # toss a coin
+        flip = random.randint(0, 1)
+        stx_opt = self._make_mount_cmd_old_or_new_style('old' if flip else 'new')
```

Comment:
This is weird. It should be part of the yaml to do the coinflip. Otherwise you could rerun a test and get different results (we normally try to maintain random seeds when scheduling tests so that it's possible to have identical configuration.

Suggest making this a configuration of the kernel mount instead so that you can use `$` and randomly chosen config fragments (overrides).

Reply:
Makes sense. I'll fix it up.